src/IECoreScene/ShaderNetworkAlgo.cpp

```diff
@@ -383,3 +385,238 @@ ShaderNetworkPtr ShaderNetworkAlgo::convertObjectVector( const ObjectVector *net
 
 	return result;
 }
+
+namespace
+{
+
+template<typename Spline>
+void expandSpline( const InternedString &name, const Spline &spline, CompoundDataMap &newParameters )
+{
+	const char *basis = "catmull-rom";
+	bool duplicateEndPoints = false;
+	if( spline.basis == Spline::Basis::bezier() )
+	{
+		basis = "bezier";
+	}
+	else if( spline.basis == Spline::Basis::bSpline() )
+	{
+		basis = "bspline";
+	}
+	else if( spline.basis == Spline::Basis::linear() )
+	{
+		duplicateEndPoints = true;
```

Comment:
I presume this is for the reason described in `OSLShader::prepareSplineCVsForOSL`? Would be good to document it...